packages/editor/src/browser/editor-manager.ts

```diff
@@ -189,6 +189,13 @@ export class EditorManager extends NavigatableWidgetOpenHandler<EditorWidget> {
         };
     }
 
+    /**
+     * Save all editors.
+     */
+    async saveAllEditors(): Promise<void> {
+        const widgets = this.shell.widgets.filter(widget => widget instanceof EditorWidget);
+        Promise.all(widgets.map(Saveable.save));
+    }
```

Comment:
We should optimize and only save editors which are dirty (this behavior is also similar with [vscode](https://github.com/microsoft/vscode/blob/cf7369db9c5d613d3825e525dd29b3c4e6c3b371/src/vs/workbench/services/editor/browser/editorService.ts#L1075-L1077)):

```ts
saveAll(options?: ISaveAllEditorsOptions): Promise<boolean> {
  return this.save(this.getAllDirtyEditors(options), options);
}
```